language/move-prover/src/prover_task_runner.rs

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! Prover task runner that runs multiple instances of the prover task and returns
+//! as soon as the fastest instance finishes.
+
+use crate::cli::Options;
+use async_std::task;
+use async_trait::async_trait;
+use futures::{future::FutureExt, pin_mut, select};
+use rand::Rng;
+use std::{
+    process::Output,
+    sync::mpsc::{channel, Sender},
+};
+use tokio::{
+    process::Command,
+    runtime::Runtime,
+    sync::{broadcast, broadcast::Receiver},
+};
+
+#[derive(Debug, Clone)]
+enum BroadcastMsg {
+    Stop,
+}
+
+#[async_trait]
+pub trait ProverTask {
```

Comment:
I just realized that this trait is closer to a ProverTaskFactory rather than a single ProverTask.